src/main/java/com/artipie/asto/fs/FileStorage.java

```diff
@@ -290,6 +293,23 @@ private Path path(final Key key) {
         return Paths.get(this.dir.toString(), key.string());
     }
 
+    /**
+     * Removes empty directories.
+     * @param key Key
+     */
+    private void deleteEmptyDirs(final Optional<Key> key) {
+        try {
+            if (key.isPresent() && !key.get().string().isEmpty()
+                && this.path(key.get()).toFile().isDirectory()
```

Comment:
@olenagerasimova you can check it with new Java API: `Files.isDirectory(path)`. `toFile()` method is needed for compatibility. Also, when using `File.isDirectory(path)` you don't need to check that `path` exist or not empty